### ISSUE TITLE
Decrypting password from [Credential] object fails with PSCore 7 on Mac

### ISSUE THREAD

**Issue by VincentBerger** [HewlettPackard/POSH-HPOneView#463]
### Expected Behavior
connect-HPOVMgmt -hostname myov -username Administrator
prompts for password and makes the connection

### Actual Behavior
Connection fails with Invalid username or password

### Steps to reproduce
I updated PowerShell Core to 7.0 on my Mac (haven't tried other platforms) and now this code no longer works:
```PowerShell
$decryptPassword = [Runtime.InteropServices.Marshal]::PtrToStringAuto([Runtime.InteropServices.Marshal]::SecureStringToBSTR($Password))
```
The resulting $decryptpassword is my password truncated to the first character 'p' and authentication fails when this is passed in the JSON body of the POST request. I suspect the PowerShell maintainers would not consider this a bug and you need to find another way to handle the password...

### Version Information
HPE OneView PowerShell Library Version: 5.0.2341.1920
Output from `$PSVersionTable`:

```PowerShell
PSVersion                      7.0.0
PSEdition                      Core
GitCommitId                    7.0.0
OS                             Darwin 18.7.0 Darwin Kernel Version 18.7.0: Thu Jan 23 06:52:12 PST 2020; root:xnu-4903.278.25~1/RELEASE_X86_64
Platform                       Unix
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by VincentBerger** [comment]
Might be exclusive to Mac though. [This closed issue](https://github.com/PowerShell/PowerShell/issues/11953) seems to match my problem

**Comment by ChrisLynchHPE** [comment]
As I have documented in the release notes, the use of `-Username` and `-Password` parameters are deprecated in the library.  While I have not removed them for not causing script backwards compatibility issues, please transition to using the `-Credential` parameter instead.  You would want to use the `Get-Credential` native Cmdlet to pass into the `-Credential` parameter like:

```powershell
Connect-HPOVMgmt -Hostname MyAppliance -Credential (Get-Credential -Username Administrator)
```

**Comment by VincentBerger** [comment]
Sure, will do. I just wanted to document it as it pops up after a PowerShell upgrade and may surprise others. Although maybe not that many others if it is indeed limited to Mac ;-)
Feel free to close this issue if you want.

**Comment by ChrisLynchHPE** [comment]
Thanks for the feedback Vincent.  I'm going to close this, as this isn't a bug for the library.

**Comment by VincentBerger** [comment]
Ahem, sorry, reopening because `Connect-HPOVMgmt -Hostname MyAppliance -Credential (Get-Credential -Username Administrator)` fails too, for exactly the same reason, since it uses the same .Net functions to convert the password to plain text before putting it in the JSON body:
```PowerShell
        elseif ($PSCmdlet.ParameterSetName -eq 'PSCredential')
        {

            $Username = $Credential.UserName
            $decryptPassword = [Runtime.InteropServices.Marshal]::PtrToStringAuto([Runtime.InteropServices.Marshal]::SecureStringToBSTR($Credential.Password))

        }
```
FWIW, changing PtrToStringAuto to PtrToStringBSTR as suggested in the PowerShell issue fixes the problem
PS: I cannot actually reopen the issue because I did not close it. Hope you see this comment anyway...

**Comment by ChrisLynchHPE** [comment]
Hmmm...  Well, I have not performed any testing with PowerShell 7.0 yet.  It is on my list to do.  And yes, I completely forgot about decrypting the password value when passing a Credential object.  There are other places where this logic happens as well, not just `Connect-HPOVMgmt` Cmdlet.

**Comment by VincentBerger** [comment]
Confirmed fixed in 5.00.2368.2592, closing
